generators/server/templates/src/main/java/package/security/oauth2/AuthRedirectController.java.ejs

```diff
@@ -0,0 +1,34 @@
+<%#
+Copyright 2013-2019 the original author or authors from the JHipster project.
+
+This file is part of the JHipster project, see https://jhipster.github.io/
+for more information.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-%>
+package <%=packageName%>.security.oauth2;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+/**
+ * This controller is protected by default, and only used to specify "/authorize" as a login-triggering link.
+ */
+@Controller
+public class AuthRedirectController {
+
+    @GetMapping("/authorize")
```

Comment:
It'd be nice if this wasn't needed. It seems you need to request a protected resource to trigger OIDC login with Spring Security 5.1. Since it comes back to this endpoint after successful authentication, it redirects to the client app.